libraries/HumanPoseFuser.js

```diff
@@ -568,33 +1015,29 @@ class HumanPoseFuser {
                 frameKey: null,
                 nodeKey: null,
                 propertyPath: 'updatedByChildren',
-                newValue: arr,
+                newValue: updatedByChildren,
                 editorId: 0
             };
         }
 
-        if (!selectedObjectId) {
-            return;
-        }
-
-        if (poseData[selectedObjectId] == undefined || poseData[selectedObjectId].joints.length == 0) {
+        if (finalPose === undefined || finalPose.joints.length == 0) {
             if (this.verbose) {
                 console.log('not updating joints: obj=' + fusedObjectId);
             }
             return;
         }
 
-        this.objectsRef[fusedObjectId].updateJoints(poseData[selectedObjectId].joints);
-        this.objectsRef[fusedObjectId].lastUpdateDataTS = poseData[selectedObjectId].timestamp;
+        this.objectsRef[fusedObjectId].updateJoints(finalPose.joints);
+        this.objectsRef[fusedObjectId].lastUpdateDataTS = finalPose.timestamp;
         server.resetObjectTimeout(fusedObjectId); // keep the object alive
 
-        // create copy and update name (confidence attribute not copied)
-        let wholePose = Object.assign({}, poseData[selectedObjectId]);
+        // create copy and update name (unnecessary properties not copied)
+        let wholePose = Object.assign({}, finalPose);
```

Comment:
I don't see any bugs resulting from this but note that Object.assign is a shallow copy, for example mutating finalPose.joints will change wholePose.joints and vice versa

Reply:
It should be fine, the pose data like joints are not expected to change one generated. In some cases, internal 'joints' array might be pointed at from several WholePoseData objects with different lifetimes. But I assume that garbage collector does a right thing and removes it once there is no reference to it.